runtime/src/tokio/runtime.rs

```diff
@@ -278,6 +279,18 @@ impl crate::Runner for Runner {
     }
 }
 
+impl AsyncExecutor for &Runner {
+    fn block_on<T>(&self, future: impl Future<Output = T>) -> T {
+        self.executor.runtime.block_on(future)
```

Comment:
A fix for bypassing the public trait is to modify `Executor::start` to be an immutable reference (so it can be called multiple times).

This would require a refactor of `deterministic` but is possible.